File: tests/expansion_test/tmp_supported_keywords.pas
program kwtest;
const a = 1;
var arr: array[1..2] of integer; i: integer;
function f(x: integer): integer;
begin
  f := x
end;
procedure p;
begin
  writeln('ok')
end;
begin
  i := 1;
  if not (i = 2) then
    case i of
      1: begin write(arr[1]); break end;
      2: write('two')
    end;
  for i := 1 to 2 do
  begin
    continue;
    read(arr[i])
  end;
  readln(i);
  p
end.

File: tests/expansion_test/tmp_supported_keywords.c
#include <stdio.h>

const int a = 1;
int arr[2];
int i;
int f(int x) {
    int f_result;
    f_result = x;
    return f_result;
}

void p() {
    printf("%s\n", "ok");
}


int main() {
    i = 1;
    if (!((i == 2))) {
        switch (i) {
            case 1:
                {
                    printf("%d", arr[1 - 1]);
                    break;
                }
                break;
            case 2:
                printf("%s", "two");
                break;
        }
    }
    for (i = 1; i <= 2; i++) {
        continue;
        scanf("%d", &arr[i - 1]);
    }
    scanf("%d", &i);
    p();
    return 0;
}
